### PR TITLE
Add RerunFailedJobsByID and RerunJobByID

### PR DESCRIPTION
**Pull request by guo-chris**
Fixes https://github.com/google/go-github/issues/2344

Add support to rerun all failed jobs or a single job from a workflow run
https://docs.github.com/en/rest/actions/workflow-runs#re-run-failed-jobs-from-a-workflow-run
https://docs.github.com/en/rest/actions/workflow-runs#re-run-a-job-from-a-workflow-run

### CODE REVIEW

**Review by gmlewis — APPROVED** [review]
Thank you, @guo-chris !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Comment by guo-chris** [comment]
Thanks @gmlewis! Do you know when this will get released after it's approved?

**Comment by gmlewis** [comment]
> Thanks @gmlewis! Do you know when this will get released after it's approved?

We are about due for another release during the second week of May.

**Comment by gmlewis** [comment]
> LGTM. Thank you for making these changes

Thank you, @plcwww !
Merging.

**Comment by gmlewis** [comment]
> Thanks @gmlewis! Do you know when this will get released after it's approved?

@guo-chris - this is now included in this release: https://github.com/google/go-github/releases/tag/v44.1.0

**Pull request merged**
